### DISCUSSION TITLE
File transfer is slow

### DISCUSSION THREAD

**Discussion by Danstiv** [BearWare/TeamTalk5#2106] [Q&A]
### Description

The speed of downloading and uploading files is slow on the client side; if there is a channel of 100 megabits, the speed reaches no more than 12 megabits.
There is no problem with this when using a custom client.

### Application

- [X] qtTeamTalk
- [ ] TeamTalkAndroid
- [ ] iTeamTalk
- [ ] TeamTalkClassic
- [ ] TeamTalkServer

Not tested with other clients

### Platform

- [X] Windows
- [ ] macOS
- [ ] Android
- [ ] iOS
- [ ] Linux

### Expected behavior

Files are transferred at a speed corresponding to the speed of the Internet.

### Actual behavior

Files are transferred at a speed several times lower than the Internet speed.

### Steps to reproduce problem

1. Prepare a test file with any content, I think 200-300 mb will be enough.

2. Download the custom client attached to this comment.
You must have python3 installed to use it, 3.7 will probably be sufficient.

3. Change the data in the last lines of the uploader.py file, substitute the data for your server and user.
For the `file_path` parameter, specify the path to your test file.

4. Make sure that the client is working, run it and wait for the upload to start; if the file starts to upload, close the client.

5. Check your internet speed, remember/write down the upload speed.

6. Try to upload some big file to tt server using QT client.
To carry out correct measurements, the server must have a good stable Internet channel.

7. Remember/write down the upload speed.

8. Run the custom client and wait for the upload to start.

9. Remember/write down the upload speed.

10. Compare the upload speeds obtained in steps 5, 7 and 9, the upload speed when using a custom client will most likely be higher and closer to your internet speed.

11. (not necessary), download the file uploaded in step 8 and make sure that it was not corrupted, I tested this client and it uploaded files without corruption, but things happen...

[file_sender.zip](https://github.com/BearWare/TeamTalk5/files/13328222/file_sender.zip)

**Comment by Danstiv** [comment]
Sorry, I accidentally uploaded a client with a Russian interface, here is the modification in English.
[file_sender.zip](https://github.com/BearWare/TeamTalk5/files/13328484/file_sender.zip)

**Comment by amirmahdifard** [comment]
i confurm this too and i also have problem with transfer speed, but i haven't reported this since i didn't think you would do anything to improve it
